changelog.d/9270.misc

```diff
@@ -0,0 +1,2 @@
+Restore PyPy compatibility by using psycopg2cffi consistently.
+Don't call `gc.freeze()` when not running on CPython for compatibility with alternative Python implementations.
```

Comment:
This should be a one-liner containing information suitable for the average synapse admin. In this case, the fine details of `gc.freeze` is probably more detail than is required.

Reply:
The PR contains two unrelated changes, how should the changelog entry be formatted in this case?

Reply:
"Fix incompatibilities with PyPy."

?

Reply:
(Anyone who cares about the *how* can click through from the changelog to the PR; we should just summarise the effect of the change).